### PR TITLE
Rsvp sort and Styling

### PR DESCRIPTION
**Pull request by dorono**
Hey @pdw207, so this is sorta messy, but I didn't want to merge your fixes to master before I had a chance to get my CSS fixes done, so for better or worse, I pulled down your branch and picked up where you left off, so both of our fixes are in this same PR, and I'm going to close the last one you opened. 

I noticed that @InciteDemocracy changed the filter design a bit w/the labels for "Sort by" and location filters being outside the buttons, so we can put that in a future PR, @InciteDemocracy, let us know if you'd like someone to go back and fix that later (or before we push to prod, for that matter, if it's really bugging you).

In summary:
* location filter added by @pdw207 
* added click handler to close "sort by" filter popup when clicking outside of the popup
* responsive css updates by me
* unit test update
* misc linting fixes

desktop-update

---

mobile-update

### CODE REVIEW

File: src/components/RsvpBadge/RsvpBadge.js [hunk 1]
```diff
@@ -6,16 +6,13 @@ const toK = (n) => {
...
+  const alignStyle = center ? styles.wrapperCenter : styles.wrapperRight;
+  const spanRSVP = <span className={styles.shortText}>{totalAccepted}+ Going</span>;
   if (type === 'mobileOnly') {
     return spanRSVP;
   }
   return (
     <div>
-      {
-        (type !== 'desktopOnly') && spanRSVP
-      }
```
Comment (dorono) [inline src/components/RsvpBadge/RsvpBadge.js]: I didn't have a chance to really investigate exactly what the `desktopOnly` and `mobileOnly` values due, but I know that we were seeing the mobile/text-based version of the RSVP count show up the "tablet" (md, I think?) and mobile widths, and removing the 3 lines above fixed the issue without breaking anything else I could find. However, It would be good if @pdw207 could confirm that I'm not breaking something else in removing these lines.

Comment (InciteDemocracy) [reply src/components/RsvpBadge/RsvpBadge.js]: This added feature is great! do you think it might be best to make to match to the styling of the mobile RSVP tags (simple grey text) for both mobile and desktop views?

Comment (dorono) [reply src/components/RsvpBadge/RsvpBadge.js]: I actually think that would be a pretty good idea, my vote is to keep them consistent. But I'd put that work into a separate ticket.

**Comment by pdw207** [comment]
Can you add `justify-content: center` to .image-and-rsvp-wrapper?

File: src/components/EventLocationFilter/EventLocationFilter.sass [hunk 2]
```diff
@@ -61,6 +61,7 @@
         flex: 1
         font-size: 0.8em
         font-weight: 700
+        text-align: left
```
[the remark this answers is not in the document]
Comment (dorono) [reply src/components/EventLocationFilter/EventLocationFilter.sass]: Hmm, I didn't see that on my end and it didn't come up in the linting, but hopefully it's gone now...

File: src/components/RsvpBadge/RsvpBadge.js [hunk 3]
```diff
@@ -6,16 +6,13 @@ const toK = (n) => {
 };
 
 const RsvpBadge = ({ totalAccepted, center, type }) => {
-  const alignStyle = center ? styles.wrapperCenter : styles.wrapperRight,
-    spanRSVP = <span className={styles.shortText}>{totalAccepted}+ Going</span>
+  const alignStyle = center ? styles.wrapperCenter : styles.wrapperRight;
+  const spanRSVP = <span className={styles.shortText}>{totalAccepted}+ Going</span>;
```
[the remark this answers is not in the document]
Comment (dorono) [reply src/components/RsvpBadge/RsvpBadge.js]: Done!

File: src/components/RsvpBadge/RsvpBadge.js (hunk 1 above)
Comment (dorono) [reply src/components/RsvpBadge/RsvpBadge.js]: Ahh, ok, looks like those lines are indeed necessary, just had to add a property to one of the RSVP tags to the EventCard component.

File: src/components/RsvpFilter/RsvpFilter.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    window.removeEventListener('click', this._handleDocumentClick);
+  }
+
+  // Close menu if clicking on the document (outside of the menu)
+  handleDocumentClick() {
+    this.setState({ rsvpModalOpen: false });
+  }
+
+  toggleModalState(e) {
+    e.stopPropagation();
```
[the remark this answers is not in the document]
Comment (dorono) [reply src/components/RsvpFilter/RsvpFilter.js]: Ahh, nice! Unfortunately, the linter was yelling at me for trying to use the syntax you have there, so I decided to submit to the will of the linter w/an if statement. :)

**Comment by dorono** [comment]
@pdw207 Yeah, I didn't have time to jump on the tests. Did you want to (Squash and) merge this in the meantime and create a new ticket for the test coverage?
